### PR TITLE
Develop

### LINKED ISSUE #4: restructure project folders
The `classes` folder should be inside of `src`.  I would renamed it to `Domain`.

I would recommend something like this:

`src/config/` for di, middleware, and routes.
`src/Action` for actions (you don't have any yet).
`src/Domain` for the current stuff you have in `classes` now.

Add:

`tests` folder in project root, and include a Bootstrap.php and SampleTest.php (see signup for example).
Also add `phpunit.xml.dist`.  

Adjust your code so that your app still functions, given the changes.

### PR DESCRIPTION
**Pull request by garcijo**
Resolved #4:restructure project folders, implemented unit tests for Web/Domain, and corrected some styling errors.

### CODE REVIEW

File: src/Web/Action/LoginAction.php [hunk 1]
```diff
@@ -28,7 +31,7 @@ public function __invoke(Request $request, Response $response, $args)
     {
         $data = $request->getParsedBody();
         $userEmail = filter_var($data['email'], FILTER_SANITIZE_STRING);
-        $userPass = filter_var($data['password'], FILTER_SANITIZE_STRING);
+        $userPass = md5(filter_var($data['password'], FILTER_SANITIZE_STRING));
```
Comment (adamaltman) [inline src/Web/Action/LoginAction.php]: `md5` maybe for a demo app, but not a proper way to handle password authentication in 2016.

File: src/Web/Action/SignupAction.php [hunk 2]
```diff
@@ -29,7 +32,7 @@ public function __invoke(Request $request, Response $response, $args)
         $data = $request->getParsedBody();
         $userName = filter_var($data['name'], FILTER_SANITIZE_STRING);
         $userEmail = filter_var($data['email'], FILTER_SANITIZE_STRING);
-        $userPass = filter_var($data['password'], FILTER_SANITIZE_STRING);
+        $userPass = md5(filter_var($data['password'], FILTER_SANITIZE_STRING));
```
Comment (adamaltman) [inline src/Web/Action/SignupAction.php]: I'll make a ticket about this...

File: src/Web/Domain/SpotifyFeed.php [hunk 3]
```diff
@@ -90,40 +96,41 @@ public function getMusic(string $user):array
...
-        $song_img = $song->album->images[1]->url;
-        $_POST['song_img'] = $song_img;
-        $song_width = $song->album->images[1]->width;
-        $_POST['song_width'] = $song_width;
+        $songLink = $song->preview_url;
+        $_POST['song_link'] = $songLink;
+        $songImg = $song->album->images[1]->url;
+        $_POST['song_img'] = $songImg;
+        $songWidth = $song->album->images[1]->width;
+        $_POST['song_width'] = $songWidth;
```
Comment (adamaltman) [inline src/Web/Domain/SpotifyFeed.php:110]: confused by why you are setting global values here?

File: tests/Web/Domain/UserMapperTest.php [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        $expectedUser = new UserEntity(['email' => $testUserName, 'name' => $testName, 'password' => $testPassword]);
+        $userMapper->createUser($testName, $testUserName, $testPassword);
+        $testLog = $userMapper->loginUser($testUserName, $testPassword);
+
+        $this->assertEquals($expectedUser, $testLog);
+
+        //remove dummy data
+        $userMapper->removeUser($testUserName);
+
+        return $testLog;
```
Comment (adamaltman) [inline tests/Web/Domain/UserMapperTest.php]: why `return` anything?

**Comment by garcijo** [comment]
resolved #12: php password. As for the global POST data, that is being read in the home page to render the music player. The code for that is in templates/home.phtml

**Pull request merged**